common/lib/src/realm_common_base.dart

```diff
@@ -16,12 +16,37 @@
 //
 // //////////////////////////////////////////////////////////////////////////////
 
+/// An enum controlling the base type for a [RealmModel].
+///
+/// {@category Annotations}
+enum ObjectType {
+  /// A standalone top-level object that can be persisted in Realm. It can link
+  /// to other objects or collections of other objects.
+  topLevel('RealmObject'),
```

Comment:
What if it was `topLevel('TopLEvel')` instead. `RealmObject` wasn't renamed to `RealmObjectBase`? It will change a lot of naming in this PR, and it can be a bit hard to evaluate without seeing it, if it would be better, but I think it might.